docroot/modules/custom/mass_migrate/src/Commands/MassMigrateCommands.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Drupal\mass_migrate\Commands;
+
+use Consolidation\AnnotatedCommand\CommandData;
+use Drush\Commands\DrushCommands;
+
+/**
+ * Mass Migrate drush commands.
+ */
+class MassMigrateCommands extends DrushCommands {
+
+  /**
+   * Override config when running drush entity:delete.
+   *
+   * @hook pre-command entity:delete
+   */
+  public function overrideEntityDeletePreCommand(CommandData $commandData) {
+    if (isset($commandData->getArgsAndOptions()['entity_type']) && isset($commandData->getArgsAndOptions()['options']['bundle'])) {
+      if ($commandData->getArgsAndOptions()['entity_type'] == 'node' && $commandData->getArgsAndOptions()['options']['bundle'] == 'service_details') {
+        // Set variables to process entity_hierarchy items with a queue
+        \Drupal::state()->set('entity_hierarchy_disable_writes', TRUE);
+      }
+    }
+  }
+
+  /**
+   * Override config when running drush entity:delete.
+   *
+   * @hook post-command entity:delete
+   */
+  public function overrideEntityDeletePostCommand() {
+    // Unset variables after running the command.
+    \Drupal::state()->set('entity_hierarchy_disable_writes', FALSE);
+  }
```

Comment:
This seems like an awkward way to set `entity_hierarchy_disable_writes`. If we leave mass_migrate enabled, we will be enabling and disabling whenever the entity:delete drush command runs. Thats true even after the migration is complete. Could we add these Drush commands to the script instead?

Reply:
I moved this over to the drush command script.